robotoff/app/api.py

```diff
@@ -1055,6 +1055,26 @@ def on_get(self, req: falcon.Request, resp: falcon.Response, username: str):
         resp.media = {"count": {"annotations": annotation_count}}
 
 
+class DisplayInsightPredictionForProducts:
+    def on_get(self, req: falcon.Request, resp: falcon.Response, barcode: str):
+        server_domain: Optional[str] = req.get_param("server_domain")
+        response: JSONType = {}
+        insights = [
+            i.serialize()
+            for i in get_insights(
+                barcode=barcode, server_domain=server_domain, limit=None
+            )
+        ]
+
+        if not insights:
+            response["status"] = "no_insights"
+        else:
+            response["insights"] = insights
+            response["status"] = "found"
+
+        resp.media = response
```

Comment:
Sorry for the ticket title was misleading.

Finally we decided to add api points akin to the one already existing for insights which is in InsightCollection

So what you have to do, is indeed make classes similar to InsightCollection (taking exactly same parameters, so you should try to refactor the parameter parsing code in a method or function), but returning predictions, images, etc.

This also probably mean you will have to do the same as get_insights function but for predictions, images etc.

Note: first do it for predictions, we shall see then after.

Reply:
Hi @alexgarel 
Made a new commit to get predictions.